### PR TITLE
feat: support tencent cos

### PR DESCRIPTION
**Pull request by leizhiyuan**
support tencent cos , test case can run in local

### CODE REVIEW

File: src/tencent/config.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    pub fn load_via_env(&self) {
+        if self.read_env.load(Ordering::Relaxed) {
+            return;
+        }
+        self.read_env.store(true, Ordering::Relaxed);
+
+        let envs = env::vars().collect::<HashMap<_, _>>();
+        let mut config = { self.config.read().expect("lock must be valid").clone() };
+
+        if let Some(v) = envs.get(TENCENT_CLOUD_ACCESS_KEY_SECRET) {
```
Comment (Xuanwo) [inline src/tencent/config.rs:59]: **question**: Do you have references that tencent cloud will use this env? If no SDKs adopt this, we should not do this either.

Comment (leizhiyuan) [reply src/tencent/config.rs:59]: OK,Previously I thought it was required to be defined by the project

**Comment by leizhiyuan** [comment]
image

I tested in my local machine with right key

**Comment by Xuanwo** [comment]
Thanks a lot! Please format code with `cargo fmt --all`, others LGTM.

File: Cargo.toml [hunk 2]
```diff
@@ -49,6 +49,9 @@ time = { version = "0.3", features = ["macros", "formatting", "parsing"] }
 ureq = { version = "2.5", default-features = false }
 rand = "0.8.5"
 rsa = { version = "0.8.1", features = ["sha2"] }
+urlencoding = "2.1.2"
+ring = "0.16.20"
```
Comment (Xuanwo) [inline Cargo.toml:53]: Why do we need to depend directly on `ring`? Can we use existing dependencies (by our best)?

**Comment by leizhiyuan** [comment]
remove all new deps

**Pull request merged**